blas/gonum/level2cmplx128.go

```diff
@@ -78,3 +79,102 @@ func (Implementation) Zgeru(m, n int, alpha complex128, x []complex128, incX int
 		kx += incX
 	}
 }
+
+// Zher performs the Hermitian rank-one operation
+//  A += alpha * x * x^H
+// where A is an n×n Hermitian matrix, alpha is a real scalar, and x is an n
+// element vector. On entry, the imaginary parts of the diagonal elements of A
+// are ignored and assumed to be zero, on return they will be set to zero.
+func (Implementation) Zher(uplo blas.Uplo, n int, alpha float64, x []complex128, incX int, a []complex128, lda int) {
+	if uplo != blas.Upper && uplo != blas.Lower {
+		panic(badUplo)
+	}
+	checkZMatrix('A', n, n, a, lda)
+	checkZVector('x', n, x, incX)
+
+	if n == 0 || alpha == 0 {
+		return
+	}
+
+	var kx int
+	if incX < 0 {
+		kx = (1 - n) * incX
+	}
+	if uplo == blas.Upper {
+		if incX == 1 {
+			for i := 0; i < n; i++ {
+				if x[i] != 0 {
+					tmp := complex(alpha*real(x[i]), alpha*imag(x[i]))
```

Comment:
Unless the benchmark is not correct, the numbers suggest to keep the row-wise operation:
```
name                      old time/op  new time/op  delta
ZherUpperN10Inc1-4         153ns ± 6%   175ns ± 5%   +14.54%  (p=0.029 n=4+4)
ZherUpperN100Inc1-4       12.8µs ± 7%   7.2µs ± 4%   -43.37%  (p=0.029 n=4+4)
ZherUpperN1000Inc1-4      1.19ms ± 9%  1.28ms ± 7%      ~     (p=0.114 n=4+4)
ZherUpperN10000Inc1-4      118ms ± 0%  1139ms ± 5%  +863.02%  (p=0.029 n=4+4)
ZherUpperN10Inc10-4        152ns ± 3%   176ns ± 3%   +15.46%  (p=0.029 n=4+4)
ZherUpperN100Inc10-4      12.4µs ± 5%   6.6µs ± 9%   -46.41%  (p=0.029 n=4+4)
ZherUpperN1000Inc10-4     1.47ms ± 6%  1.55ms ± 7%      ~     (p=0.343 n=4+4)
ZherUpperN10000Inc10-4     236ms ± 3%  1204ms ± 4%  +409.24%  (p=0.029 n=4+4)
ZherUpperN10Inc1000-4      149ns ± 7%   180ns ± 4%   +20.77%  (p=0.029 n=4+4)
ZherUpperN100Inc1000-4    12.8µs ± 4%   8.0µs ± 2%   -37.46%  (p=0.029 n=4+4)
ZherUpperN1000Inc1000-4   1.46ms ± 1%  2.32ms ± 6%   +59.43%  (p=0.029 n=4+4)
ZherUpperN10000Inc1000-4   747ms ± 6%  1741ms ± 1%  +133.18%  (p=0.029 n=4+4)
ZherLowerN10Inc1-4         147ns ± 6%   169ns ± 8%   +15.33%  (p=0.029 n=4+4)
ZherLowerN100Inc1-4       12.3µs ± 3%   7.1µs ± 8%   -42.15%  (p=0.029 n=4+4)
ZherLowerN1000Inc1-4      1.25ms ± 7%  1.30ms ± 9%      ~     (p=0.686 n=4+4)
ZherLowerN10000Inc1-4      121ms ± 4%  1101ms ± 6%  +810.65%  (p=0.029 n=4+4)
ZherLowerN10Inc10-4        143ns ± 8%   170ns ± 4%   +18.50%  (p=0.029 n=4+4)
ZherLowerN100Inc10-4      12.5µs ± 7%   6.9µs ±10%   -44.83%  (p=0.029 n=4+4)
ZherLowerN1000Inc10-4     1.45ms ± 4%  1.53ms ± 3%      ~     (p=0.057 n=4+4)
ZherLowerN10000Inc10-4     234ms ± 4%  1129ms ± 1%  +381.79%  (p=0.029 n=4+4)
ZherLowerN10Inc1000-4      140ns ±11%   162ns ±10%      ~     (p=0.057 n=4+4)
ZherLowerN100Inc1000-4    12.7µs ± 5%   8.0µs ± 4%   -37.02%  (p=0.029 n=4+4)
ZherLowerN1000Inc1000-4   1.65ms ± 7%  2.43ms ± 5%   +47.40%  (p=0.029 n=4+4)
ZherLowerN10000Inc1000-4   813ms ± 2%  1650ms ± 1%  +103.03%  (p=0.029 n=4+4)
```
old is row-wise with for loops, new is column-wise with `c128.AxpyInc`.